### DISCUSSION TITLE
Dcukdb don't seems to be using all CPU core ?

### DISCUSSION THREAD

**Discussion by djouallah** [duckdb/duckdb#4659] [Q&A] [answered]
just testing a cloud function using this code

```
import duckdb 
import pandas as pd
con = duckdb.connect()
SQL = '''
with xx(i) as ( SELECT * FROM range(0, 1000, 1) )
select i*random()/200000000 as y  from xx order by y desc limit 10;
    '''
df =con.execute(SQL).df()
df
```
I see a low CPU usage, isn't sorting a multi threaded operation?

**Comment by lnkuiper** [comment, marked as answer]
The `range` table function is a single-threaded source, so there will be (virtually) no parallelism for this query. Try creating a table from the `range` first, then running your query, and you should see parallelism.

Furthermore, DuckDB parallelizes per 100k rows. So in general, any query that reads less than 100k rows will not be executed in parallel.

**Reply by djouallah** [reply]
not bad at all !!!
![image](https://user-images.githubusercontent.com/12554469/189316941-312d1c51-7d6d-4393-8327-c4dc14d18f9c.png)
